packages/core-types/tests/integration/components/field-editors/date-editor-test.js

```diff
@@ -1,34 +1,64 @@
 import { module, test } from 'qunit';
 import { setupRenderingTest } from 'ember-qunit';
-import { render } from '@ember/test-helpers';
+import { render, triggerEvent } from '@ember/test-helpers';
+import EmberObject from '@ember/object';
 import hbs from 'htmlbars-inline-precompile';
 
 module('Integration | Component | field editors/date editor', function(hooks) {
   setupRenderingTest(hooks);
 
   test('it renders', async function(assert) {
-    this.set('model', {
+    this.set('model', EmberObject.create({
       expiration: '2030-01-01'
-    });
-    await render(hbs`{{field-editors/date-editor content=model field="expiration"}}`);
+    }));
+    await render(hbs`
+      {{field-editors/date-editor
+        content=model
+        field="expiration"
+      }}`);
     assert.dom('input[type=date]').hasValue('2030-01-01', 'date input has correct value');
-    assert.dom('input').isNotDisabled('date field is not disabled');
+    assert.dom('input').isNotDisabled();
   });
 
-  test('it renders with invalid date', async function(assert) {
+  test('it renders with an invalid date', async function(assert) {
     this.set('model', {
       expiration: 'pizza'
     });
-    await render(hbs`{{field-editors/date-editor content=model field="expiration"}}`);
+    await render(hbs`
+      {{field-editors/date-editor
+        content=model
+        field="expiration"
+      }}`);
     assert.dom('input[type=date]').hasNoValue('date input has no value');
-    assert.dom('input').isNotDisabled('date field is not disabled');
+    assert.dom('input').isNotDisabled();
   });
 
   test('it can be disabled', async function(assert) {
-    this.set('model', {
+    this.set('model', EmberObject.create({
       expiration: '2030-01-01'
+    }));
+    await render(hbs`
+      {{field-editors/date-editor
+        content=model
+        field="expiration"
+        disabled=true
+      }}`);
+    assert.dom('input').isDisabled();
+  });
+
+  test('onchange is called when the field is left', async function(assert) {
+    let changed;
+    this.set('onchange', () => {
+      changed = true;
     });
```

Comment:
an alternative pattern is this:
```js
this.set('onchange', () => assert.step('change'));
// ...
assert.verifySteps(['change']);
```

this has the advantage or verifying that the action was only called exactly once

Reply:
I love that. It was passive knowledge but it's so much better.